test/system/shoot_cp_migration/migrate_test.go

```diff
@@ -202,28 +209,30 @@ func afterMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp app
 	if err := t.PopulateAfterMigrationComparisonElements(ctx); err != nil {
 		return err
 	}
+	if !t.Shoot.Status.IsHibernated {
```

Comment:
Nit: you can move the code that has to be executed in both cases (not hibernated and hibernated shoot) here, then just return nil if `Shoot.Status.IsHibernated` is true and then add the checks that require the shoot to not be hibernated.